processor/schemaprocessor/internal/translation/revision.go

```diff
@@ -0,0 +1,190 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/translation"
+
+import "go.opentelemetry.io/otel/schema/v1.0/ast"
+
+// Revision represents all changes that are to be
+// applied to a signal at a given version.
+type Revision interface {
+	// Version is stored to help make
+	// sorting revisions easier
+	Version() *Version
+
+	// All is intended to apply to all incoming attributes
+	// that are known as part of this revision.
+	All() Modifier
```

Comment:
Is this not just a combination of all the others? Where is this used?

Reply:
So this is a like for like from the `1.0.0` format, https://opentelemetry.io/docs/reference/specification/schemas/file_format_v1.0.0/#all-section

Basically, if the schemaURL matches then the `all` grouping is applied to every signal.